### PR TITLE
24 interface solver

### PR DESCRIPTION
**Pull request by Mitchelmore**

### CODE REVIEW

File: src/Components/HeaderSolver.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import AppBar from "@mui/material/AppBar";
+import Button from "@mui/material/Button";
+import Toolbar from "@mui/material/Toolbar";
+import Typography from "@mui/material/Typography";
+
+type HeaderProps = {
+  title: string
+}
+
+export const Header = ({ title }: HeaderProps) => {
```
Comment (SKroell) [inline src/Components/HeaderSolver.tsx:10]: What is the difference between using this and simply using the Header.tsx? They look very similar

File: src/Components/HelpSolver.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export const HelpSolver = () => {
```
Comment (SKroell) [inline src/Components/HelpSolver.tsx:1]: We can export different components from the same file, should we maybe keep all Helper components in Help.tsx?

File: src/Pages/Solver.tsx [hunk 3]
```diff
@@ -1,19 +1,213 @@
...
+      let graph = parser.parse(e.target.value);
+      this.setState({parseError: ""})
+    } catch (err: any) {
+      this.setState({parseError: (err.message + "</br>" + JSON.stringify(err.location))})
+    }
+
+  }
+
+  // Adds a new symbol/activity mapping to the exercise
+  addSymbolFields() {
```
Comment (SKroell) [inline src/Pages/Solver.tsx:77]: Not sure this is needed, this will alter the exercise? Solver should be read-only

File: src/Pages/Solver.tsx [hunk 4]
```diff
@@ -1,19 +1,213 @@
...
+    symbols.push({ symbol: "", activity: "" });
+    this.setState({ exercise: { ...this.state.exercise, symbols: symbols } });
+  }
+
+  // Removes a symbol/activity mapping from the exercise
+  removeSymbolFields(index: number) {
+    let symbols = this.state.exercise.symbols;
+    symbols.splice(index, 1);
+    this.setState({ exercise: { ...this.state.exercise, symbols: symbols } });
+  }
```
Comment (SKroell) [inline src/Pages/Solver.tsx:88]: Not sure this is needed, this will alter the exercise? Solver should be read-only

**Review by SKroell — CHANGES_REQUESTED** [review]

File: src/Components/HeaderSolver.tsx (hunk 1 above)
Comment (Mitchelmore) [reply src/Components/HeaderSolver.tsx:10]: Gives the option to make the headers different. We talked about highlighting the open page in the header.

File: src/Components/HelpSolver.tsx (hunk 2 above)
Comment (Mitchelmore) [reply src/Components/HelpSolver.tsx:1]: Good idea. I'll look at it tomorrow

File: src/Pages/Solver.tsx (hunk 3 above)
Comment (Mitchelmore) [reply src/Pages/Solver.tsx:77]: This and below is the functionality that might be needed to read in the data from the json file (the buttons are gone) - so they are present as functions. Agree that Solver should be read-only. My thought is that reading in data could be similar in the Creator and Solver. If another solution works, fine to choose that.

File: src/Pages/Solver.tsx (hunk 4 above)
Comment (Mitchelmore) [reply src/Pages/Solver.tsx:88]: Yes, this one should probably be removed (reading in the next question can start again and add lines as required or whatever is used).

File: src/Components/HeaderSolver.tsx (hunk 1 above)
Comment (Mitchelmore) [reply src/Components/HeaderSolver.tsx]: The HeaderSolver.tsx has been deleted.

File: src/Components/HelpSolver.tsx (hunk 2 above)
Comment (Mitchelmore) [reply src/Components/HelpSolver.tsx]: The HelpSolver.tsx file has been deleted, and the HelpSolver code is now in Help.tsx

File: src/Pages/Solver.tsx (hunk 3 above)
Comment (Mitchelmore) [reply src/Pages/Solver.tsx:80]: Keeping this for now, but have included comments in the code.

File: src/Pages/Solver.tsx (hunk 4 above)
Comment (Mitchelmore) [reply src/Pages/Solver.tsx]: This has been removed.

**Pull request merged**